src/feature/hs/hs_common.c

```diff
@@ -1723,45 +1714,45 @@ hs_get_extend_info_from_lspecs(const smartlist_t *lspecs,
     }
   } SMARTLIST_FOREACH_END(ls);
 
+  fascist_firewall_choose_address_ls(lspecs, FIREWALL_OR_CONNECTION, 0, &ap);
+
   /* Legacy ID is mandatory, and we require IPv4. */
```

Comment:
fascist_firewall_choose_address_ls() can return an IPv6 address, so we require an IP address.
(We only require IPv4 for connections that are not direct.)